### DISCUSSION TITLE
Is dcm4che affected by log4j security issues?

### DISCUSSION THREAD

**Discussion by jehos** [dcm4che/dcm4chee-arc-light#3491] [Q&A] [answered]
Log4j has recently encountered a security issue. Log4j before version 2.14 is affected.

Dcm4chee-arc-light is also affected?
And which version is safe?

> CVE-2021-44228
> The Log4j team has been made aware of a security vulnerability, CVE-2021-44228, that has been addressed in Log4j 2.15.0.
> 
> Log4j’s JNDI support has not restricted what names could be resolved. Some protocols are unsafe or can allow remote code execution. Log4j now limits the protocols by default to only java, ldap, and ldaps and limits the ldap protocols to only accessing Java primitive objects by default served on the local host.
> 
> One vector that allowed exposure to this vulnerability was Log4j’s allowance of Lookups to appear in log messages. As of Log4j 2.15.0 this feature is now disabled by default. While an option has been provided to enable Lookups in this fashion, users are strongly discouraged from enabling it.
> 
> For those who cannot upgrade to 2.15.0, in releases >=2.10, this vulnerability can be mitigated by setting either the system property log4j2.formatMsgNoLookups or the environment variable LOG4J_FORMAT_MSG_NO_LOOKUPS to true. For releases from 2.7 through 2.14.1 all PatternLayout patterns can be modified to specify the message converter as %m{nnolookups} instead of just %m. For releases from 2.0-beta9 to 2.10.0, the mitigation is to remove the JndiLookup class from the classpath:zip -q -d log4j-core-*.jar org/apache/logging/log4j/core/lookup/JndiLookup.class

**Comment by gunterze** [comment, marked as answer]
No.
- dcm4che does not use log4j2. 
- Wildfly does not use Log4j2 by default configuration - s.o. http://www.mastertheboss.com/jbossas/jboss-log/how-to-use-log4j2-in-your-wildfly-applications/ -  so the archive and the keycloak container is not affected.
- For ELK containers, you may set
  ```
    environment:
      LOG4J_FORMAT_MSG_NO_LOOKUPS: "true"
  ```
  in your docker-compose.yaml, or generally
  ```
  LOG4J_FORMAT_MSG_NO_LOOKUPS=true
  ```
  in `env_file: docker-compose.env`.
  
  s. https://www.docker.com/blog/apache-log4j-2-cve-2021-44228/
